lib/openscap_parser/profile.rb

```diff
@@ -46,6 +46,12 @@ def selected_entity_ids
         @parsed_xml.xpath("select[@selected='true']/@idref").map(&:text)
     end
 
+    def refined_value_ids
+      @refined_value_ids ||= @parsed_xml.xpath("refine-value").each_with_object({}) do |element, rv|
+        rv[element.at_xpath('@idref').text] = element.at_xpath('@selector').text
+      end
```

Comment:
This sets up a hash that looks like `{idref1: selector1, idref2: selector2, idref3: selector3, ....}`
The key is the id of the `value` that's being refined and value of the hash is the `@selector` attribute and this will match one of the `@selector` attributes on the Value definition for the value that's being refined. This is how we find out what value the Profile wants to use.

Reply:
Is it okay that this method returns a hash where the key is the `idref` of a Value and the value is the `selector` associated with a value?

In the `refine-value` element in the Profile definition, the only information we have to use is the `idref` of a Value and the `selector` that is associated with the value:
`<refine-value idref="xccdf_org.ssgproject.content_value_var_selinux_policy_name" selector="targeted"/>`

So this `refined_values` method won't give us exactly what we need for this in compliance-backend:
`Profile#value_refinemens:json -> {ref_id: value, ref_id: value}`
We'll need to replace the `selectors` in the hash with the actual `value`. I don't think this would be too complicated though right?

Reply:
I think the hash should work :+1: